backend/validators/greyboxing.lua

```diff
@@ -13,79 +14,86 @@ function get_mock_response_from_table(table, message)
 	return "no match"
 end
 
-function just_in_backend(args, prompt, format)
-	local search_in = lib.select_backends_for(args, format)
-
-	lib.log{ args = args, prompt = prompt, format = format, search_in = search_in }
-
-	local handler = args.handler
+local function just_in_backend(ctx, args, mapped_prompt)
+	---@cast mapped_prompt MappedPrompt
+	---@cast args LLMExecPromptPayload | LLMExecPromptTemplatePayload
 
 	-- Return mock response if it exists
-	if args.host_data and args.host_data.mock_response then
-		if args.payload then
-			if args.payload.principle then
-				-- Return the matching response to gl.eq_principle_prompt_comparative request which contains a principle key in the payload
-				result = get_mock_response_from_table(args.host_data.mock_response.eq_principle_prompt_comparative, prompt.user_message)
-			elseif args.payload.output then
-				-- Return the matching response to gl.eq_principle_prompt_non_comparative request which contains an output key in the payload
-				result = get_mock_response_from_table(args.host_data.mock_response.eq_principle_prompt_non_comparative, prompt.user_message)
-			else
-				-- Return the matching response to gl.exec_prompt request which does not contain any specific key in the payload
-				result = get_mock_response_from_table(args.host_data.mock_response.response, prompt.user_message)
-			end
+	if ctx.host_data.mock_response then
```

Comment:
@kstroobants Is there a code that tests this part? I am not 100% sure I patched this mocks correctly